pkg/hive/feature_lifecycle.go

```diff
@@ -123,5 +124,5 @@ func (fl *FeatureLifecycle) List() []Feature {
 	fl.mu.Lock()
 	defer fl.mu.Unlock()
 
-	return maps.Keys(fl.hooks)
+	return append([]Feature{}, slices.Collect(maps.Keys(fl.hooks))...)
```

Comment:
Is the `append` really needed? Only difference I can see is that this makes sure a `nil` slice isn't returned, but that's totally fine. I'd fix the tests if that's what's breaking.

Reply:
Yes, its mainly to avoid breaking current tests. I will remove this and fix the tests accordingly. Thanks